src/flatdirectory.ts

```diff
@@ -28,25 +28,28 @@ const defaultCallback: DownloadCallback = {
 export class FlatDirectory {
     private ethStorageRpc?: string;
     private contractAddr?: string;
-    private chainId!: number;
-    private retries!: number;
 
+    private chainId!: number;
     private wallet!: ethers.Wallet;
     private blobUploader!: BlobUploader;
 
+    private retries: number = MAX_RETRIES;
     public isSupportBlob: boolean = false;
 
     static async create(config: SDKConfig) {
         const flatDirectory = new FlatDirectory();
-        await flatDirectory.init(config);
+        if (config.privateKey) {
+            await flatDirectory.init(config);
+        } else {
+            await flatDirectory.initWithoutPrivateKey(config);
```

Comment:
Why do we need a separate init method?

Reply:
A separate init method provides better distinction and functional separation. The standard init process requires initializing many components.

Reply:
I mean can we call `initWithoutPrivateKey` (maybe we should rename the func) in the `init` to reduce the duplicated code